### PR TITLE
Fix inconsistency between CHF and Servlet API cookie handling. #22

### PR DESCRIPTION
**Pull request by pavelhoral**
This should be backported to sustaining/22.0.x after merging to master.

### CODE REVIEW

File: auth-filters/forgerock-authn-filter/forgerock-jaspi-modules/forgerock-jaspi-jwt-session-module/src/main/java/org/forgerock/jaspi/modules/session/jwt/AbstractJwtSessionModule.java [hunk 1]
```diff
@@ -564,11 +573,11 @@ private String buildJwtString(JwtClaimsSet claimsSet, Key publicKey) {
      * @param now The Date at which the cookie was created.
      * @param exp The expiry Date of the cookie.
      */
-    private int getCookieMaxAge(Date now, Date exp) {
+    private Integer getCookieMaxAge(Date now, Date exp) {
```
Comment (karelmaxa) [inline auth-filters/forgerock-authn-filter/forgerock-jaspi-modules/forgerock-jaspi-jwt-session-module/src/main/java/org/forgerock/jaspi/modules/session/jwt/AbstractJwtSessionModule.java:576]: Javadoc of this method is not valid anymore. Could you please reword it?

**Review by karelmaxa — CHANGES_REQUESTED** [review]

**Comment by pavelhoral** [comment]
Changed the behaviour back to servlet and grizzly style (i.e. max-age is either negative for session, zero to actually delete the cookie and positive for a standard expiration). So CHF is the odd one out, but that is OK.

File: auth-filters/forgerock-authn-filter/forgerock-jaspi-modules/forgerock-jaspi-jwt-session-module/src/main/java/org/forgerock/jaspi/modules/session/jwt/AbstractJwtSessionModule.java (hunk 1 above)
Comment (pavelhoral) [reply auth-filters/forgerock-authn-filter/forgerock-jaspi-modules/forgerock-jaspi-jwt-session-module/src/main/java/org/forgerock/jaspi/modules/session/jwt/AbstractJwtSessionModule.java]: Changed back to previous semantic (see comment bellow).

**Comment by pavelhoral** [comment]
SemaphoreCI is spreading fake news :) - the build is OK. Not sure what happened there.

**Review by karelmaxa — APPROVED** [review]
LGTM.

**Pull request merged**